draft-gpew-priv-ppm.md

```diff
@@ -672,247 +678,381 @@ independently.
 
-The AggregateReq request is used by the leader to send a set of reports to the
-helper. These reports MUST all be associated with the same PPM task and batch.
+The leader begins aggregation by choosing a set of candidate reports that pertain
+to the same PPM task. The leader can run this process for many candidate reports
+in parallel as needed. After choosing the set of candidates, the leader begins
+aggregation by splitting each report into "report shares", one for each aggregator.
+The leader and helpers then run the aggregate initialization flow to accomplish two tasks:
 
-For each aggregator endpoint `[aggregator]` in `AggregateReq.task_id`'s
-parameters except its own, the leader sends a POST request to
-`[aggregator]/aggregate` with the following message:
+1. Recover and determine which input report shares are invalid.
+1. For each valid report share, initialize the VDAF preparation process.
+
+An invalid report share is marked with one of the following errors:
 
 ~~~
-struct {
-  TaskID task_id;
-  opaque agg_param<0..2^16-1>;  // VDAF aggregation parameter
-  opaque helper_state<0..2^16>; // helper's opaque state
-  AggregateSubReq seq<1..2^24-1>;
-} AggregateReq;
+enum {
+  batch-collected(0),
+  report-replayed(1),
+  report-dropped(2),
+  hpke-unknown-config-id(3),
+  hpke-decrypt-error(4),
+  vdaf-prep-error(5),
+} ReportShareError;
 ~~~
 
-The structure contains the PPM task, an opaque, VDAF-specific aggregation
-parameter, an opaque *helper state* string, and a sequence of *sub-requests*,
-each corresponding to a unique client report. Sub-requests are structured as
-follows:
+The leader and helper initialization behavior is detailed below.
+
+#### Leader Initialization
+
+The leader begins the aggregate initialization phase with the set of candidate report
+shares as follows:
+
+1. Decrypt the input share for each report share as described in {{input-share-decryption}}.
+1. Check that the resulting input share is valid as described in {{input-share-batch-validation}}.
+1. Initialize VDAF preparation as described in {{input-share-prep}}.
+
+If any step yields an invalid report share, the leader removes the report share from
+the set of candidate reports. Once the leader has initialized this state for all valid
+candidate report shares, it then creates an AggregateInitReq message for each helper to
+initialize the preparation of this candidate set. The AggregateInitReq message is
+structured as follows:
 
 ~~~
 struct {
-  Nonce nonce;                     // Equal to Report.nonce.
-  Extension extensions<0..2^16-1>; // Equal to Report.extensions.
-  HpkeCiphertext helper_share;
-  opaque message<0..2^16-1>; // VDAF message
-} AggregateSubReq;
+  Nonce nonce;
+  Extension extensions<0..2^16-1>;
+  HpkeCiphertext encrypted_input_share;
+} ReportShare;
+
+struct {
+  TaskID task_id;
+  opaque agg_param<0..2^16-1>;
+  opaque helper_state<0..2^16>;
+  ReportShare report_shares<1..2^16-1>;
+} AggregateInitReq;
 ~~~
```

Comment:
```suggestion
~~~

[[OPEN ISSUE: consider sending report shares separately (in parallel) to the aggregate instructions. RIght now, aggregation parameters and the corresponding report shares are sent at the same time, but this may not be strictly necessary. ]]
```